### PR TITLE
Spring core

### PR DESCRIPTION
**Pull request by vyark**

### CODE REVIEW

File: src/main/java/com/epam/facade/BookingFacadeImpl.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+import java.util.List;
+
+@Component
+public class BookingFacadeImpl implements BookingFacade {
+
+    private UserService userService;
+    private TicketService ticketService;
+    private EventService eventService;
+
+    public BookingFacadeImpl(UserService userService, TicketService ticketService,
```
Comment (fedorchuck) [inline src/main/java/com/epam/facade/BookingFacadeImpl.java]: can we use lombock?

File: src/main/java/com/epam/service/UserService.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    public User getUserByEmail(String email) {
+        return userDao.values().stream()
+                      .filter(user -> user.getEmail().equals(email))
+                      .findFirst()
+                      .orElseThrow(() -> new IllegalStateException("User does not exist with " +
+                              "email: " + email));
+    }
+
+    public List<User> getUsersByName(String name, int pageSize, int pageNum) {
+        int skipCount = (pageNum - 1) * pageSize;
```
Comment (fedorchuck) [inline src/main/java/com/epam/service/UserService.java:32]: magic number? btw, why minus 1, not 3?

File: src/test/java/com/epam/dao/TicketDaoTest.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.sql.Date;
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.epam.storage.Storage.*;
```
Comment (fedorchuck) [inline src/test/java/com/epam/dao/TicketDaoTest.java]: :-)

Comment (fedorchuck) [inline src/test/java/com/epam/dao/TicketDaoTest.java]: if you want, I can show how to disable it on idea

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-test-autoconfigure</artifactId>
+            <version>${org.springframework.autoconfigure.version}</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>javax.annotation</groupId>
+            <artifactId>javax.annotation-api</artifactId>
+            <version>${javax.annotation-api.version}</version>
```
Comment (fedorchuck) [inline pom.xml:58]: ordering or scope, some below

**Review by fedorchuck — COMMENTED** [review]
I can accept it but I'd suggest to make the changes

File: src/main/java/com/epam/facade/BookingFacadeImpl.java (hunk 1 above)
Comment (vyark) [reply src/main/java/com/epam/facade/BookingFacadeImpl.java]: Replaced

File: src/main/java/com/epam/service/UserService.java (hunk 2 above)
Comment (vyark) [reply src/main/java/com/epam/service/UserService.java:32]: We calculate the number of users to skip for pagination.

pageNum=1, pageSize=5, skipCount=0
pageNum=2, pageSize=5, skipCount=5
pageNum=3, pageSize=5, skipCount=10
pageNum=4, pageSize=5, skipCount=15

File: pom.xml (hunk 4 above)
Comment (vyark) [reply pom.xml:58]: could you please clarify?

**Review by fedorchuck — APPROVED** [review]

**Pull request merged**